repos/system_upgrade/el7toel8/actors/kernel/checkkerneldrivers/libraries/library.py

```diff
@@ -0,0 +1,34 @@
+import os
+
+from leapp.exceptions import StopActorExecutionError
+from leapp.libraries.stdlib import api
+from leapp.models import PCIDevices
+
+
+def get_removed_drivers(path):
+    removed = set()
+    try:
+        with open(path, 'r') as f:
+            # Extracting kernel drivers from the file.
+            for line in f.readlines():
+                token = line.strip()
+                if token.startswith('#') or not token:
+                    # We do not want comments or empty lines.
+                    continue
+                removed.add(token)
+        return removed
+    except (IOError, OSError) as e:
+        raise StopActorExecutionError('Cannot read {}: {}'.format(os.path.abspath(path), str(e)))
+
+
+def get_present_drivers():
+    present = set()
+    for fact in api.consume(PCIDevices):
```

Comment:
Nit: 
```
devices = [d for f in api.consume(PCIDevices) for d in f.devices if d.driver]
return set(devices)
```

Reply:
Good idea, even though this seems a bit opaque. And I'm collecting drivers, not devices.

Reply:
Replaced with a set comprehension.